src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/utils/metadata/MetadataSchema.java

```diff
@@ -0,0 +1,54 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.utils.metadata;
+
+import org.openkilda.model.bitops.BitField;
+
+public class MetadataSchema {
+    public static final MetadataSchema INSTANCE = new MetadataSchema();
+
+    private static final BitField LLDP_MARKER_FLAG         = new BitField(0x0000_0000_0000_0001L);
+    private static final BitField ARP_MARKER_FLAG          = new BitField(0x0000_0000_0000_0004L);
+
+    private static final BitField ONE_SWITCH_FLOW_FLAG     = new BitField(0x0000_0000_0000_0002L);
+
+    private static final BitField OUTER_VLAN_FIELD         = new BitField(0x0000_0000_0000_fff0L);
+    private static final BitField OUTER_VLAN_PRESENCE_FLAG = new BitField(0x1000_0000_0000_0000L);
```

Comment:
Why don't we just use `0x0000_0000_0000_0008L` as a flag?
This will save us from using `HALF_SIZE_METADATA` feature.

Reply:
I have a doubts regarding usage of bits in upper half of metadata. I prefer to have tools that allow us to work with all possible hardware, also I want to highlight, that some hardware do not implement OF spec requirement and has such "limitation".

One more argument - is a requirement to make migration if we will decide to change any bit/flag location/meaning.

Reply:
I have moded the outer-VLAN presence flag into the lower half of metadata schema.